kernel/tests/v2_checkpoints.rs

```diff
@@ -0,0 +1,236 @@
+// rustfmt::skip
+use std::sync::Arc;
+
+use arrow_array::RecordBatch;
+use delta_kernel::engine::sync::SyncEngine;
+
+use delta_kernel::engine::arrow_data::ArrowEngineData;
+use delta_kernel::{DeltaResult, Table};
+
+mod common;
+use common::{load_test_data, read_scan};
+
+fn read_v2_checkpoint_table(test_name: impl AsRef<str>) -> DeltaResult<Vec<RecordBatch>> {
+    let test_dir = load_test_data("tests/data", test_name.as_ref()).unwrap();
+    let test_path = test_dir.path().join(test_name.as_ref());
+
+    let table = Table::try_from_uri(test_path.to_str().expect("table path to string")).unwrap();
+    let engine = Arc::new(SyncEngine::new());
+    let snapshot = table.snapshot(engine.as_ref(), None)?;
+    let scan = snapshot.into_scan_builder().build()?;
+    let batches = read_scan(&scan, engine)?;
+
+    Ok(batches)
+}
+
+fn test_v2_checkpoints(table_name: &str) -> DeltaResult<()> {
+    let num_of_max_digits = 999.to_string().len();
+    let batches = read_v2_checkpoint_table(table_name)?;
+
+    fn generate_rows(count: usize, width: usize) -> Vec<String> {
+        (0..count)
+            .map(|id| format!("| {:<width$} |", id, width = width))
+            .collect()
+    }
```

Comment:
Given that width is always `999.to_string().len()` = 3, why not do:
```suggestion
        let max_width = 3;
        (0..count)
                 .map(|id| format!("| {:<width$} |", id, width = max_width))
            .collect()
    }
```

Reply:
I had gone with the prior to include some context but I can swap it out if you think it's unnecessary 👍